docs/architecture-decisions/2023-06-06_frontend-templates.md

```diff
@@ -13,23 +13,54 @@ AlgoKit v2 aims provide an end-to-end development and deployment experience that
-- **Maintainability**: The dApp templating feature should be easy to maintain and extend. This implies that the feature should be implemented in a way that allows for easy addition of new templates and/or modification of existing ones as the complexity and variety of templates scale.
-- **Seamless onramp**: The dApp templating feature should provide a seamless onramp for developers to get started with dApp development. This implies that the feature should provide a simple and intuitive way to get started with dApp development and deployment. Providing developers a choice on whether they want more flexibility or rely on default recommended practices.
+### Independent Frontend Templates
+
+1. Create official frontend template that complies with AlgoKit's principles of modularity, maintainability, and flexibility while also serving as a reference for template builders.
+
+2. Expand algokit functionality to provide a clear process for developers to link frontend templates with the typed clients generated by backend templates.
+
+3. Design frontend templates such that backend and frontend have less than zero dependencies on each other, ensuring high modularity.
+
+### End-to-End Starter Repositories
+
+1. Establish a starter repository that allows developers to bootstrap end-to-end dApp projects within a single command, while also serving as an example for template builders on how to efficiently couple their backend and frontend templates.
+2. Implement a bundling process for backend and frontend templates in the starter repositories that code duplication and maintenance overhead.
+3. Develop deployment pipelines for frontend components inside end-to-end starter repository that supports TestNet, and MainNet deployments to hosting providers of choice (Netlify, Vercel and etc to be decided).
+4. Ensure that the end-to-end starter repository supports both manual deployment and deployment via a continuous deployment pipeline.
 
 ## Principles
 
 - [AlgoKit Guiding Principles](../../docs/algokit.md#Guiding-Principles) - all of the aforementioned requirements should be met in a way that is consistent with the guiding principles of AlgoKit or attempt to find a balanced trade of between the principles that satisfies the requirements.
 
-## Options
+## Explored options
+
+Enhancing AlgoKit's templating capabilities involves a notable shift from managing individual templates to co-locating multiple templates or projects together. This is a marked increase in complexity that requires careful consideration of the varying trade-offs between different approaches. The subsequent sections explore these options and their associated trade-offs, providing a concise overview of this new challenge and our proposed solutions.
+
+### Option 1: Monorepos
```

Comment:
I'm not quite sure this is the right name for this option.

Monorepo implies single repository, multiple projects.

The description below sounds more like "Monolithic template" to me?